### ISSUE TITLE
duplicate lists of IP's in Telegram messages

### ISSUE THREAD

**Issue by revelt** [theMiddleBlue/log2iptables#2]
Hi @theMiddleBlue 
I don't know is it me or is it something wrong with the system.
I just tried the code from your Telegram example to my crontab, appending only Telegram commands from another example:

```
*/5 * * * * /usr/local/bin/log2iptables.sh -x 1 -f /var/log/auth.log -r "sshd.*(f|F)ail.*(\=| )([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})" -p 3 -l 5 -t 1 -T "<replacedtoken>" -C "<replacedCharID>" > /dev/null 2>&1
*/1 * * * * /usr/local/bin/log2iptables.sh -x 1 -f /var/log/syslog -r "PortScan.*SRC\=([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)" -p 1 -l 1 > /dev/null 2>&1
```

Telegram does work, but it sends me _identical_ messages with the same list of blocked IP's:

<img width="224" alt="exhibit_a" src="https://cloud.githubusercontent.com/assets/8344688/14158109/4abf9be2-f6c6-11e5-953d-c3a04b048f21.png">

However, what's strange is that running the same code straight from the crontab code (without `*/5 * * * *` and `-t 1 -T ...onwards`) does yield new IP's:

<img width="504" alt="exhibit_b" src="https://cloud.githubusercontent.com/assets/8344688/14158161/9bb55636-f6c6-11e5-9d59-47e688ec1b04.png">

I would expect that those three addresses from the above would get reported via Telegram.

thanks upfront, Roy

**Comment by theMiddleBlue** [comment]
Hi @revelt 

sorry for the late answer. The right behaviour should be that when log2iptables found a match, it notifies you when new IP address(es) are added to an IPTables chain, and sending you a detail about all **new** added IP addresses. When it start again, it should skip all IP Addresses already present on an IPTables chain and notifies you about new entries only.

Could you confirm that the first cron-job correctly adds to your IPTables all the IP Addresses found by log2iptables? 

i've try with the following example log file:
```
# cat testl.log
127.0.12.1
127.0.12.2
127.0.12.3
```

the first time i run log2iptables with this command:
```sh
./log2iptables.sh -x 1 -f test.log -r '([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)' -p 1 -l 1 -t 1 -T "<mytoken>" -C <chatid>
```

![schermata 2016-03-31 alle 14 51 50](https://cloud.githubusercontent.com/assets/4454961/14176229/471d4054-f750-11e5-917d-936bd7709c6f.png)

and i've correctly received the following Telegram message:
![schermata 2016-03-31 alle 14 49 43](https://cloud.githubusercontent.com/assets/4454961/14176253/5dfb6e90-f750-11e5-9c75-25fcde799e38.png)

the second and third time that i run the same command:
![schermata 2016-03-31 alle 14 54 30](https://cloud.githubusercontent.com/assets/4454961/14176293/8472389c-f750-11e5-9564-e0be7960d654.png)

and it has not send me any message.

thanks!

-theMiddle

**Comment by revelt** [comment]
Very strange, it looks like my cron-job doesn't do anything at all. Apparently, all ip's that were added to _iptables_ were a result of manual commands I ran on `log2iptables`, not a result of cronjob using the `log2iptables`. I discovered this by deleting one entry in the _iptables_ and waiting for couple cycles of cronjob to complete — the IP entry I removed was not added by cronjob. I'll close this issue now and try to troubleshoot crontab further. Thanks for help!
